index_impl.go

```diff
@@ -28,8 +28,28 @@ import (
 	"strings"
 )
 
+func indexStringToType(indexString string) (IndexType, error) {
+	switch indexString {
+	case string(FullTextIndex):
+		return FullTextIndex, nil
+	case string(HashIndex):
+		return HashIndex, nil
+	case string(SkipListIndex):
+		return SkipListIndex, nil
+	case string(PrimaryIndex):
+		return PrimaryIndex, nil
+	case string(PersistentIndex):
+		return PersistentIndex, nil
+	case string(GeoIndex), "geo1", "geo2":
+		return GeoIndex, nil
+
+	default:
+		return "", WithStack(InvalidArgumentError{Message: "unknown index type"})
+	}
+}
+
 // newIndex creates a new Index implementation.
-func newIndex(id string, col *collection) (Index, error) {
+func newIndex(id string, indexString string, col *collection) (Index, error) {
```

Comment:
No big issues, but `indexString` is really meaningless. Consider changing to `indexTypeStr`